INACTIVE: ZOS Xlp Codecache and ObjectHeap Changes

AlenBadel opened the pull request:
Z/OS now supports the same command line argument format as other architectures for both -Xlp:codecache, and -Xlp:objectheap.

The behaviour is slightly different with the objectheap, where pagrable large files are always preferred over nonpageable when [non]pageable is not specified.

Signed-off-by: AlenBadel <Alen.Badel@ibm.com>

AlenBadel commented:
Related OpenJ9 PR
https://github.com/eclipse/openj9/pull/7585

pshipton commented on docs/xlpobjectheap.md:
```diff
@@ -31,14 +31,10 @@ To find out the large page sizes available and the current setting, use the `-ve
 
-z/OS&reg;:
-
-        -Xlp:objectheap:pagesize=<size>[,strict|warn][,pageable|nonpageable]
```
This still needs to be a valid option on z/OS, and shouldn't be removed from the docs.

pshipton replied:
I note this option specification in incorrect until eclipse/openj9#7585 is merged, as pageable|nonpageable is not optional

pshipton commented on docs/xlpobjectheap.md:
```diff
@@ -61,6 +57,8 @@ See [Using -X command-line options](x_jvm_commands.md) for more information abou
 
 : On macOS, the default page size is 4 KB.
 
+: On Z/OS&reg If both 1M pageable, and nonpageable large pages are available then 1M pageable large pages will be used by default if [non]pageable is not specified.
```
When the pageable|nonpageable option isn't specified ...

doveye commented:
There doesn't seem to be a corresponding doc issue for this PR. Please can you create one - updates should be committed against that. The content team can then offer help. For example:
- The What's New topic (https://eclipse.github.io/openj9-docs/version0.18/) also needs updating to advise customers of the change
- For -Xlp:codecache, it sounds like we are introducing another way for z/OS customers to do the same thing, ie:
`-Xlp:codecache:pagesize=<size>`==`-Xlp:codecache:pagesize=<size>,pageable`
If so, we should mention the old option only to say it is still recognised. Will it be deprecated? (If support is added for nonpageable in the future, `pageable|nonpageable` can be treated as a new suboption?) In the Default Values section, you could omit mentioning either option since the behaviour is the same, but I think the entry needs clarifying anyway as it's currently quite confusing. It says 

> 1 MB pageable pages, when available, are the default size for the code cache.

and also 

> The ... option supports only a large page size of 1 MB pageable large pages. 

Both sentences could be replaced with "The default size for the code cache is 1 MB pageable pages. No other large page sizes are supported."
- For -Xlp:objectheap, the docs already say that `pageable|nonpageable` is optional? The `pageable|nonpageable` section could be modified to say something like:

> On z/OS systems, defines the type of memory to allocate for the Java object heap, if you want to override the default value.

> Supported pageable sizes: 1 MB and 4 KB.
> Supported nonpageable sizes: 2 GB and 1 MB.

SueChaplain commented:
@AlenBadel - all docs changes require an issue to be opened for tracking and historical purposes. Please follow this link: https://github.com/eclipse/openj9-docs/issues/new?template=new-documentation-change.md

SueChaplain commented:
@AlenBadel - all docs changes require an issue to be opened for tracking and historical purposes. Please follow this link: https://github.com/eclipse/openj9-docs/issues/new?template=new-documentation-change.md

pmhayward commented:
@AlenBadel ... Hi Alen, would be grateful if you'd create a doc issue for this work, outlining what was required - so that we can track it with other 0.18.0 changes. Then we can get this finished off and merged.
Many thanks, Peter

AlenBadel commented:
@pmhayward My apologies, I've added the issue.

https://github.com/eclipse/openj9-docs/issues/445

pmhayward commented:
@AlenBadel ... Many thanks

pmhayward commented:
@AlenBadel ... Alen, just realised that the commit you made a short while ago wasn't signed off. Therefore Eclipse is currently rejecting the PR. If you open the "Details" link under "All checks have failed ", it gives more details and something about how to fix it. Thanks, Peter

AlenBadel commented:
@pmhayward I'm currently working on addressing all the comments given. This PR is dependant on https://github.com/eclipse/openj9/pull/7585, so it can't be merged until that PR is accepted.

pmhayward commented:
@AlenBadel  I'm assuming that as the openj9 PR is still open, this docs PR should **not** currently be merged; please confirm! Are these changes still likely to happen at some point? Thanks!

SueChaplain commented:
I propose to close this PR as it is very old with respect to the current code base. If these changes are still planned, please rebase and raise a new PR.

SueChaplain commented:
Closing. Commented in #445 to determine whether a future PR is expected or whether we can close the issue.
